### PR TITLE
Added fetching of branches from main repository during update task.

### LINKED ISSUE #153: `gulp update` could do a `fetch all` in all repos (including the core)
I've seen that I miss such thing. I run `gulp update` and then I expect that all repos are fully fetched, that I don't have to do anything more there if I want to switch to some other branch.

### PR DESCRIPTION
**Pull request by oskarwrobel**
Fixes #153

### CODE REVIEW

**Comment by Reinmar** [comment]
I found a bug. When I change ckeditor5's package.json to point to ckeditor5-engine to `#t/xxx` and run gulp update this happens:


```
[11:58:25] Checking out ckeditor5-engine to t/xxx...

error: pathspec 't/xxx' did not match any file(s) known to git.

[11:58:25] 'update' errored after 5.19 s
[11:58:25] Error: Error while executing cd /www/ckeditor5/ckeditor5-engine && git checkout t/xxx: error: pathspec 't/xxx' did not match any file(s) known to git.

    at Object.shExec (/www/ckeditor5/ckeditor5/dev/tasks/dev/utils/tools.js:42:10)
    at Object.checkout (/www/ckeditor5/ckeditor5/dev/tasks/dev/utils/git.js:80:9)
    at module.exports.symlinks.filter.dir.forEach (/www/ckeditor5/ckeditor5/dev/tasks/dev/tasks/update.js:52:9)
    at Gulp.<anonymous> (/www/ckeditor5/ckeditor5/dev/tasks/dev/tasks.js:47:3)
    at module.exports (/www/ckeditor5/ckeditor5/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/www/ckeditor5/ckeditor5/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/www/ckeditor5/ckeditor5/node_modules/orchestrator/index.js:214:10)
    at Gulp.Orchestrator.start (/www/ckeditor5/ckeditor5/node_modules/orchestrator/index.js:134:8)
    at /usr/local/lib/node_modules/gulp/bin/gulp.js:129:20
    at nextTickCallbackWith0Args (node.js:452:9)
    at process._tickCallback (node.js:381:13)
    at Function.Module.runMain (module.js:449:11)
    at startup (node.js:139:18)
    at node.js:999:3
```

**Comment by Reinmar** [comment]
The above does not occur when the repo is already on the specified branch before pulling changes.

**Comment by oskarwrobel** [comment]
Ok, thanks. I think I know why it happens.

**Comment by oskarwrobel** [comment]
Fixed.. Sorry for stupid bug.

**Comment by Reinmar** [comment]
BTW. We'll need to upgrade [wiki article](https://github.com/ckeditor/ckeditor5/wiki/Development-Workflow) after this changes. Could you handle that as well? I also remember that there were some comments in the code that may need to be changed. Please ask @szymonkups.

**Comment by oskarwrobel** [comment]
Ok.

File: dev/tasks/dev/utils/git.js [hunk 1]
```diff
@@ -97,6 +96,20 @@ module.exports = {
 	},
 
 	/**
+	 * Fetch all branches from each origin on selected repository.
+	 *
+	 * @param {String} repositoryLocation
+	 */
+	fetchAll( repositoryLocation ) {
```
Comment (szymonkups) [inline dev/tasks/dev/utils/git.js]: Please add a test for this function in `dev/tests/dev/git.js` 
image

**Pull request merged**